src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -64,24 +68,32 @@ private void processBlob(String blobName, String containerName) {
                 .getBlobContainerClient(containerName)
                 .getBlobClient(blobName);
 
-            leaseClient = leaseClientProvider.get(blobClient);
-
-            leaseClient.acquireLease(60);
-            byte[] rawBlob = tryToDownloadBlob(blobClient);
-
-            dispatcher.dispatch(blobName, rawBlob, containerName);
-            UUID envelopeId = envelopeRepository.insert(createNewEnvelope(
-                blobName,
-                containerName,
-                blobClient.getProperties().getCreationTime().toInstant()
-            ));
-
-            logger.info(
-                "Finished processing {} from {} container. New envelope ID: {}",
-                blobName,
-                containerName,
-                envelopeId
-            );
+            if (this.readinessChecker.isReady(blobClient.getProperties().getCreationTime().toInstant())) {
```

Comment:
We could assign creation time to local variable as it is used twice in the same scope. Will also make if statement a little bit more sensible

Reply:
good point